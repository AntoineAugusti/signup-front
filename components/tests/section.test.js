
/* eslint no-unused-vars: "error" */
import ReactTestRenderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import Section from '../section'

import Container from '../../container'

describe('components | Section', () => {
  describe('render', () => {
    const renderer = ReactTestRenderer.create(<Section />)
    it('should be defined', () => {
      expect(renderer).toBeDefined()
    })
    it('should match snapshot', () => {
      expect(renderer).toMatchSnapshot()
    })
  })

  describe('Container', () => {
    const props = {
      title: 'The title',
      children: '<p>Some html</p>'
    }
    it('should have proper title', () => {
      const wrapper = shallow(<Section {...props} />)
      const container = wrapper.find(Container)
      expect(container.props().children).toEqual([<h2>The title</h2>, props.children])
    })
  })
})
