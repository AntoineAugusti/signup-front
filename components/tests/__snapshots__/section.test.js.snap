// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Section render should match snapshot 1`] = `
<section
  className="className"
  id={null}
>
  <h2
    className="section__title"
  >
    The title
  </h2>
  &lt;p&gt;Some html&lt;/p&gt;
</section>
`;
